### PR TITLE
feat: archive notes

### PR DESCRIPTION
**Pull request by bharati-21**
This PR contains the following features-
- archive a note
- unarchive a note
- edit an archived note
- delete an archived note
- function - updateArchiveNotesController added in the ArchiveController to edit archived notes
- new route to edit archived notes /api/archives/:noteId

### CODE REVIEW

File: src/reducers/notesReducerFunction.js [hunk 1]
```diff
@@ -3,15 +3,17 @@ const actionTypes = {
 	SET_NOTES_ERROR: "SET_NOTES_ERROR",
 	RESET_NOTES: "RESET_NOTES",
 	SHOW_NEW_NOTE_FORM: "SHOW_NEW_NOTE_FORM",
-	EDIT_NOTE: "EDIT_NOTE",
+	SET_ARCHIVES: "SET_ARCHIVES",
+	EDIT_ARCHIVES: "EDIT_ARCHIVES",
```
Comment (ravikumar0403) [inline src/reducers/notesReducerFunction.js]: Usually these constants are stored in a separate file named actions.js

**Review by ravikumar0403 — APPROVED** [review]
Do add links of deploy previews. The code looks clean and easy to understand, I liked the folder structure

**Review by yashpatel1797 — APPROVED** [review]
the file structure is good.
code looks clean.
if possible you can provide a deploy link, screenshot or demo video of the design.

File: src/reducers/notesReducerFunction.js (hunk 1 above)
Comment (bharati-21) [reply src/reducers/notesReducerFunction.js]: Yes, I have moved them.

**Comment by bharati-21** [comment]
> Do add links of deploy previews. The code looks clean and easy to understand, I liked the folder structure

Yes, I am yet to deploy. Shall do it before raising the next PR.
Thank you for your review!

**Comment by bharati-21** [comment]
> the file structure is good. code looks clean. if possible you can provide a deploy link, screenshot or demo video of the design.

Yes, I shall include the deploy preview. Yet to deploy the app/
Thank you for your review!

**Pull request merged**